src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -116,39 +87,32 @@ public static CollectorManager createHybridCollectorManager(final SearchContext
             searchContext.from(0);
         }
 
-        Weight filteringWeight = null;
-        // Check for post filter to create weight for filter query and later use that weight in the search workflow
-        if (Objects.nonNull(searchContext.parsedPostFilter()) && Objects.nonNull(searchContext.parsedPostFilter().query())) {
-            Query filterQuery = searchContext.parsedPostFilter().query();
-            ContextIndexSearcher searcher = searchContext.searcher();
-            // ScoreMode COMPLETE_NO_SCORES will be passed as post_filter does not contribute in scoring. COMPLETE_NO_SCORES means it is not
-            // a scoring clause
-            // Boost factor 1f is taken because if boost is multiplicative of 1 then it means "no boost"
-            // Previously this code in OpenSearch looked like
-            // https://github.com/opensearch-project/OpenSearch/commit/36a5cf8f35e5cbaa1ff857b5a5db8c02edc1a187
-            filteringWeight = searcher.createWeight(searcher.rewrite(filterQuery), ScoreMode.COMPLETE_NO_SCORES, boostFactor);
-        }
-
-        return searchContext.shouldUseConcurrentSearch()
-            ? new HybridCollectorConcurrentSearchManager(
-                numDocs,
-                new HitsThresholdChecker(Math.max(numDocs, searchContext.trackTotalHitsUpTo())),
-                trackTotalHitsUpTo,
-                filteringWeight,
-                searchContext
-            )
-            : new HybridCollectorNonConcurrentManager(
-                numDocs,
-                new HitsThresholdChecker(Math.max(numDocs, searchContext.trackTotalHitsUpTo())),
-                trackTotalHitsUpTo,
-                filteringWeight,
-                searchContext
-            );
+        // Earlier we use to initiate Collector manager on the basis of concurrent search/non-concurrent search.
+        // This we use to do because createHybridCollectorManager was called from HybridAggregationProcessor. But now we don't call it from
+        // there.
+        // 1. createHybridCollectorManager is called from HybridQueryCollectorContextSpec
+        // 2. HybridQueryCollectorContextSpec is created from the DefaultQueryPhaseSearcher [searchWithCollector
+        // method](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/query/QueryPhase.java#L441).
```

Comment:
Comment looks good. 
One suggestion: We should link the code from the released branch rather main for stability purpose and below too